amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutRecordsPublisher.java

```diff
@@ -313,8 +331,33 @@ private void errorOccurred(RecordFlow triggeringFlow, Throwable t) {
         }
     }
 
-    private void clearRecordsDeliveryQueue() {
-        recordsDeliveryQueue.clear();
+    // This method is not thread safe. This needs to be executed after acquiring lock on this.lockObject
+    private void resetRecordsDeliveryStateOnSubscriptionInit() {
+        // Clear any lingering records in the queue.
+        if (!recordsDeliveryQueue.isEmpty()) {
+            recordsDeliveryQueue.clear();
+        }
+        // Set pendingActiveSubscriptionShutdown to default value.
+        pendingActiveSubscriptionShutdown = false;
```

Comment:
Should we log warning if pendingActiveSubscriptionShutdown is found true, or there are records in the queue? 

Either of these cases points to something not working as expected, and logging status under such conditions can help us for RC'ing issues.

Reply:
yep, they will be informational.